Makefile

```diff
@@ -1,85 +1,100 @@
+## Formatting = UPPERCASE
+ARCH = x86_64
+VERSION = 39
+IMAGE_REPO = ghcr.io/ublue-os
+IMAGE_NAME = base-main
+IMAGE_TAG = $(version)
+VARIANT = Silverblue
+WEB_UI = false
+
+# Generated vars
+## Formatting = _UPPERCASE
+_BASE_DIR = $(shell pwd)
+_IMAGE_REPO_ESCAPED = $(subst /,\/,$(IMAGE_REPO))
+_IMAGE_REPO_DOUBLE_ESCAPED = $(subst \,\\\,$(_IMAGE_REPO_ESCAPED))
+
+ifeq ($(VARIANT),'Server')
+_LORAX_ARGS = --macboot --noupgrade
 else
-lorax_args = --nomacboot
+_LORAX_ARGS = --nomacboot
 endif
 
-ifeq ($(web_ui),true)
-lorax_args += -i anaconda-webui
+ifeq ($(WEB_UI),true)
+_LORAX_ARGS += -i anaconda-webui
 endif
 
-$(image_name)-$(version).iso: boot.iso container/$(image_name)-$(image_tag) xorriso/input.txt
-	xorriso -dialog on < $(base_dir)/xorriso/input.txt
+# Step 6: Build end ISO file
+## Default action
+$(IMAGE_NAME)-$(IMAGE_TAG).iso: output/$(IMAGE_NAME)-$(IMAGE_TAG).iso
+	mv output/$(IMAGE_NAME)-$(IMAGE_TAG).iso $(IMAGE_NAME)-$(IMAGE_TAG).iso
 
+output/$(IMAGE_NAME)-$(IMAGE_TAG).iso: boot.iso container/$(IMAGE_NAME)-$(IMAGE_TAG) xorriso/input.txt
+	mkdir $(_BASE_DIR)/output
+	xorriso -dialog on < $(_BASE_DIR)/xorriso/input.txt
+
+# Step 2: Build boot.iso using Lorax
 boot.iso: lorax_templates/set_installer.tmpl lorax_templates/configure_upgrades.tmpl
-	rm -Rf $(base_dir)/results
-	lorax -p $(image_name) -v $(version) -r $(version) -t $(variant) \
-          --isfinal --buildarch=$(arch) --volid=$(image_name)-$(arch)-$(version) \
-          $(lorax_args) \
+	rm -Rf $(_BASE_DIR)/results
+	lorax -p $(IMAGE_NAME) -v $(VERSION) -r $(VERSION) -t $(VARIANT) \
+          --isfinal --buildarch=$(ARCH) --volid=$(IMAGE_NAME)-$(ARCH)-$(VERSION) \
+          $(_LORAX_ARGS) \
           --repo /etc/yum.repos.d/fedora.repo \
           --repo /etc/yum.repos.d/fedora-updates.repo \
-          --add-template $(base_dir)/lorax_templates/set_installer.tmpl \
-		  --add-template $(base_dir)/lorax_templates/configure_upgrades.tmpl \
-          $(base_dir)/results/
-	mv $(base_dir)/results/images/boot.iso $(base_dir)/
+          --add-template $(_BASE_DIR)/lorax_templates/set_installer.tmpl \
+		  --add-template $(_BASE_DIR)/lorax_templates/configure_upgrades.tmpl \
+          $(_BASE_DIR)/results/
+	mv $(_BASE_DIR)/results/images/boot.iso $(_BASE_DIR)/
 
-container/$(image_name)-$(image_tag):
+# Step 3: Download container image
+container/$(IMAGE_NAME)-$(IMAGE_TAG):
 	mkdir container
-	podman pull $(image_repo)/$(image_name):$(image_tag)
-	podman save --format oci-dir -o $(base_dir)/container/$(image_name)-$(image_tag) $(image_repo)/$(image_name):$(image_tag)
-	podman rmi $(image_repo)/$(image_name):$(image_tag)
+	podman pull $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
+	podman save --format oci-dir -o $(_BASE_DIR)/container/$(IMAGE_NAME)-$(IMAGE_TAG) $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
+	podman rmi $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
 
 install-deps:
 	dnf install -y lorax xorriso podman git rpm-ostree
 
 
-
+# Step 1: Generate Lorax Templates
```

Comment:
Same with moving Step #1 to the top. Unless this messes with the benefits of using a Makefile